### PR TITLE
pfring: convert to plugin - v3

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: #11476 

Changes:
- add upgrade notes
- add ci job
- include headers in dist (caught by ci job)

Ticket: https://redmine.openinfosecfoundation.org/issues/7162

### CODE REVIEW

**Review by victorjulien — APPROVED** [review]

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
  

Pipeline 21493

**Comment by jasonish** [comment]
> ERROR: QA failed on SURI_TLPR1_alerts_cmp.

Sync issue? This PR shouldn't really change anything in QA.

**Comment by victorjulien** [comment]
> > ERROR: QA failed on SURI_TLPR1_alerts_cmp.
> 
> Sync issue? This PR shouldn't really change anything in QA.

You can set the `waiting for qa` label to trigger a rerun. I did that just now.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .uptime | 642 | 673 | 104.83% | 

Pipeline 21509

File: plugins/pfring/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# PF_RING Plugin Capture Plugin
+
+## Building
+
+To build this plugin, built Suricata with the `--enable-pfring` and
```
Comment (jlucovsky) [inline plugins/pfring/README.md:5]: nit: s/built/build

File: doc/userguide/upgrade/8.0-pfring-plugin.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+``pfring.so`` to the ``plugins`` section of your ``suricata.yaml``
+manually.
+
+Then your existing PF_RING command line options and configuration
+should continue to work.
+
+Caveats
+-------
+
+Currently builing the PF_RING plugin is not compatible with the
```
Comment (jlucovsky) [inline doc/userguide/upgrade/8.0-pfring-plugin.rst:41]: nit: s/builing/building/

**Comment by victorjulien** [comment]
Merged in #11506, thanks!

**Pull request closed without merging**